File: front-end/src/app/types/restaurant.ts
import type { Promotion } from "./promotion";

export interface Restaurant {
  id: number;
  name: string;
  rating: number;
  distance: string;
  dietary: string;
  cuisine: string;
  isOpen: boolean;
  type: "gold" | "red";
  /** [longitude, latitude] for map display */
  coordinates: [number, number];
  image?: string;
  promotions?: Promotion[];
}

export interface SearchPreferences {
  cuisine: string;
  priceRange: string;
  dietary: string;
  distance: string;
  ambience: string;
  time: string;
}

export interface SearchHistoryEntry {
  id: string;
  query: string;
  date: string;
  preferences: Partial<SearchPreferences>;
}

export interface UserProfileData {
  displayName: string;
  email: string;
  avatarUrl?: string;
  savedPreferences: SearchPreferences | null;
  searchHistory: SearchHistoryEntry[];
  favoriteRestaurants: Restaurant[];
}

File: back-end/src/database/schemas/restaurant.py
from pydantic import BaseModel, Field


class RestaurantResponse(BaseModel):
    id: int
    name: str
    rating: float
    distance: str
    dietary: str
    cuisine: str
    isOpen: bool
    type: str
    coordinates: list[float]
    image: str | None = None

    model_config = {"from_attributes": True}


class RestaurantSearchParams(BaseModel):
    cuisine: str | None = None
    dietary: str | None = None
    price_range: str | None = None
    ambience: str | None = None
    time: str | None = None
    query: str | None = None


class SearchPreferencesSchema(BaseModel):
    cuisine: str = ""
    priceRange: str = ""
    dietary: str = ""
    distance: str = ""
    ambience: str = ""
    time: str = ""


class SearchHistoryCreate(BaseModel):
    query: str
    preferences: SearchPreferencesSchema = Field(default_factory=SearchPreferencesSchema)


class SearchHistoryResponse(BaseModel):
    id: str
    query: str
    date: str
    preferences: SearchPreferencesSchema
